src/spouts/rss/feed.php

```diff
@@ -103,35 +100,42 @@ public function getContent() {
     }
 
     public function getIcon() {
-        if ($this->faviconUrl !== null) {
-            return $this->faviconUrl;
-        }
+        return null;
+    }
 
+    public function getSourceIcon() {
         // Try to use feed logo first
         $feedLogoUrl = $this->feed->getImageUrl();
-        if ($feedLogoUrl && $this->imageHelper->fetchFavicon($feedLogoUrl)) {
-            $this->faviconUrl = $this->imageHelper->getFaviconUrl();
-            $this->logger->debug('icon: using feed logo: ' . $this->faviconUrl);
+        if ($feedLogoUrl && ($iconData = $this->imageHelper->fetchFavicon($feedLogoUrl)) !== null) {
+            list($faviconUrl, $iconBlob) = $iconData;
 
-            return $this->faviconUrl;
+            $aspectRatio = $iconBlob->getWidth() / $iconBlob->getHeight();
+
+            if (0.8 < $aspectRatio && $aspectRatio < 1.3) {
+                $this->logger->debug('icon: using feed logo: ' . $faviconUrl);
+
+                return $faviconUrl;
```

Comment:
Eventually, we will want to return the blob as well so that it does not have to be downloaded again but we have already decreased the number of downloads a lot and I do not want to delay this any longer.